components/core/config/schemas.txt

```diff
@@ -1,19 +1,97 @@
-// Delimiters
-delimiters: \t\r\n!"#$%&'\(\)\*,:;<>?@\[\]\^_`\{\|\}~
+// ===
+// Timestamps
+// ===
+
+// E.g. 2015-01-31T15:50:45.392
+timestamp:\d{4}\-\d{2}\-\d{2}T\d{2}:\d{2}:\d{2}.\d{3}
+
+// E.g. 2015-01-31T15:50:45,392
+timestamp:\d{4}\-\d{2}\-\d{2}T\d{2}:\d{2}:\d{2},\d{3}
+
+// E.g. [2015-01-31T15:50:45
+timestamp:\[\d{4}\-\d{2}\-\d{2}T\d{2}:\d{2}:\d{2}
+
+// E.g. [20170106-16:56:41]
+timestamp:\[\d{4}\d{2}\d{2}\-\d{2}:\d{2}:\d{2}\]
 
-// Timestamps (using the `timestamp` keyword)
 // E.g. 2015-01-31 15:50:45,392
+// E.g. INFO [main] 2015-01-31 15:50:45,085
+timestamp:\d{4}\-\d{2}\-\d{2} \d{2}:\d{2}:\d{2},\d{3}
+
 // E.g. 2015-01-31 15:50:45.392
+timestamp:\d{4}\-\d{2}\-\d{2} \d{2}:\d{2}:\d{2}.\d{3}
+
+// E.g. [2015-01-31 15:50:45,085]
+timestamp:\[\d{4}\-\d{2}\-\d{2} \d{2}:\d{2}:\d{2},\d{3}\]
+
 // E.g. 2015-01-31 15:50:45
-timestamp:\d{4}\-\d{2}\-\d{2} \d{2}:\d{2}:\d{2}(\.\d{3}){0,1}
-// E.g. [20150131-15:50:45]
-timestamp:\[\d{8}\-\d{2}:\d{2}:\d{2}\]
+// E.g. Started POST /api/v3/internal/allowed for 127.0.0.1 at 2017-06-18 00:20:44
+// E.g. update-alternatives 2015-01-31 15:50:45
+timestamp:\d{4}\-\d{2}\-\d{2} \d{2}:\d{2}:\d{2}
+
+// E.g. Start-Date: 2015-01-31  15:50:45
+timestamp:\d{4}\-\d{2}\-\d{2}  \d{2}:\d{2}:\d{2}
+
+// E.g. 2015/01/31 15:50:45
+timestamp:\d{4}/\d{2}/\d{2} \d{2}:\d{2}:\d{2}
+
+// E.g. 15/01/31 15:50:45
+timestamp:\d{2}/\d{2}/\d{2} \d{2}:\d{2}:\d{2}
+
+// E.g. 150131  9:50:45
+timestamp:\d{2}\d{2}\d{2} [ 0-9]{2}:\d{2}:\d{2}
+
+// E.g. 01 Jan 2016 15:50:17,085
+timestamp:\d{2} [A-Z][a-z]{2} \d{4} \d{2}:\d{2}:\d{2},\d{3}
+
+// E.g. Jan 01, 2016 3:50:17 PM
+timestamp:[A-Z][a-z]{2} \d{2}, \d{4} [ 0-9]{2}:\d{2}:\d{2} [AP]M
 
+// E.g. January 31, 2015 15:50
+timestamp:[A-Z][a-z]+ \d{2}, \d{4} \d{2}:\d{2}
+
+// E.g. E [31/Jan/2015:15:50:45
+// E.g. localhost - - [01/Jan/2016:15:50:17
+// E.g. 192.168.4.5 - - [01/Jan/2016:15:50:17
+timestamp:\[\d{2}/[A-Z][a-z]{2}/\d{4}:\d{2}:\d{2}:\d{2}
+
+// E.g. 192.168.4.5 - - [01/01/2016:15:50:17
+timestamp:\[\d{2}/\d{2}/\d{4}:\d{2}:\d{2}:\d{2}
+
+// E.g. ERROR: apport (pid 4557) Sun Jan  1 15:50:45 2015
+timestamp:[A-Z][a-z]{2} [A-Z][a-z]{2} [ 0-9]{2} \d{2}:\d{2}:\d{2} \d{4}
+
+// E.g. <<<2016-11-10 03:02:29:936
+timestamp:\<\<\<\d{4}\-\d{2}\-\d{2} \d{2}:\d{2}:\d{2}:\d{3}
+
+// E.g. Jan 21 11:56:42
+timestamp:[A-Z][a-z]{2} \d{2} \d{2}:\d{2}:\d{2}
+
+// E.g. 01-21 11:56:42.392
+timestamp:\d{2}\-\d{2} \d{2}:\d{2}:\d{2}.\d{3}
+
+// E.g. 2016-05-08 11:34:04.083464
+timestamp:\d{4}\-\d{2}\-\d{2} \d{2}:\d{2}:\d{2}.\d{6}
+
+// ===
+// Delimiters
+// ===
+
+delimiters: \t\r\n!"#$%&'\(\)\*,:;\<\>\?@\[\]\^_`\{\|\}~=
+
+// ===
 // Specially-encoded variables (using the `int` and `float` keywords)
+// ===
+
 int:\-{0,1}[0-9]+
 float:\-{0,1}[0-9]+\.[0-9]+
 
+// ===
 // Dictionary variables
+// ===
+
 hex:[a-fA-F]+
+
+// Note: the way `equals` and `hasNumber` are written only work if `=` is a delim
```

Comment:
@SharafMohamed Can you clarify this? I can sort of imagine something since `equals` has `=` in it, but I don't understand why `hasNumber` won't work without `=` as a delim.